contrib/gotests_templates/function.tmpl

```diff
@@ -0,0 +1,107 @@
+{{define "function"}}
+{{- $f := .}}
+
+func {{.TestName}}(t *testing.T) {
+	{{- with .Receiver}}
+		{{- if .IsStruct}}
+			{{- if .Fields}}
+				type fields struct {
+				{{- range .Fields}}
+					{{Field .}} {{.Type}}
+				{{- end}}
+				}
+			{{- end}}
+		{{- end}}
+	{{- end}}
+	{{- if .TestParameters}}
+	type args struct {
+		{{- range .TestParameters}}
+				{{Param .}} {{.Type}}
+		{{- end}}
+	}
+	{{- end}}
+	tests := []struct {
+		name string
+		{{- with .Receiver}}
+			{{- if and .IsStruct .Fields}}
+				fields fields
+			{{- else}}
+				{{Receiver .}} {{.Type}}
+			{{- end}}
+		{{- end}}
+		{{- if .TestParameters}}
+			args args
+		{{- end}}
+		setup    func() error // Optional, delete if unused
+		teardown func() error // Optional, delete if unused
```

Comment:
nit: Do we really want to return an error here? It's always skipped, since `teardown()` is always running with `defer`.

Reply:
This is nice since we can know that resources aren't being cleaned up, etc during one test that might effect others.

Reply:
Then again, we can log the output and not care about the error.